src/ray/raylet/scheduling_queue.cc

```diff
@@ -6,49 +6,90 @@ namespace ray {
 
 namespace raylet {
 
+bool SchedulingQueue::TaskQueue::Append(const TaskID &task_id, const Task &task) {
+  if (map_.find(task_id) != map_.end()) {
+    // This shouldn't happen.
```

Comment:
if this shouldn't happen, we need to `RAY_CHECK` and fail hard if it does happen. Basically, any time you think something shouldn't happen -- it's an assumption that we must have an assert for.

Reply:
revised. thanks.